pkg/capabilities/consensus/ocr3/aggregators/identical.go

```diff
@@ -0,0 +1,152 @@
+	count           int
+}
+
+var _ types.Aggregator = (*identicalAggregator)(nil)
+
+func (a *identicalAggregator) Aggregate(_ *types.AggregationOutcome, observations map[ocrcommon.OracleID][]values.Value, f int) (*types.AggregationOutcome, error) {
+	counters := []map[[32]byte]*counter{}
+	for i := 0; i < a.config.ExpectedObservationsLen; i++ {
+		counters = append(counters, map[[32]byte]*counter{})
+	}
+	for nodeID, nodeObservations := range observations {
+		if len(nodeObservations) == 0 || nodeObservations[0] == nil {
+			a.lggr.Warnf("node %d contributed with empty observations", nodeID)
+			continue
+		}
+		if len(nodeObservations) != a.config.ExpectedObservationsLen {
+			a.lggr.Warnf("node %d contributed with an incorrect number of observations %d - ignoring them", nodeID, len(nodeObservations))
+			continue
+		}
+		for idx, observation := range nodeObservations {
+			marshalled, err := proto.MarshalOptions{Deterministic: true}.Marshal(values.Proto(observation))
+			if err != nil {
+				return nil, err
+			}
+			sha := sha256.Sum256(marshalled)
+			elem, ok := counters[idx][sha]
+			if !ok {
+				counters[idx][sha] = &counter{
+					fullObservation: observation,
+					count:           1,
+				}
+			} else {
+				elem.count++
+			}
+		}
+	}
+	return a.collectHighestCounts(counters, f)
+}
+
+func (a *identicalAggregator) collectHighestCounts(counters []map[[32]byte]*counter, f int) (*types.AggregationOutcome, error) {
+	useOverrides := len(a.config.KeyOverrides) == len(counters)
+	outcome := make(map[string]any)
+	var encoderName string
+	var encoderConfig *pb.Map
+	for idx, shaToCounter := range counters {
+		highestCount := 0
+		var highestObservation values.Value
+		for _, counter := range shaToCounter {
+			if counter.count > highestCount {
+				highestCount = counter.count
+				highestObservation = counter.fullObservation
+			}
+		}
+		if highestCount < 2*f+1 {
+			return nil, fmt.Errorf("can't reach consensus on observations with index %d", idx)
+		}
+		if useOverrides {
+			outcome[a.config.KeyOverrides[idx]] = highestObservation
+		} else {
+			outcome[fmt.Sprintf("%d", idx)] = highestObservation
+		}
+		if a.config.OverrideEncoder {
+			if idx == a.config.EncoderNameObservationIndex {
+				// extract encoder name from observation
+				name, ok := highestObservation.(*values.String)
+				if !ok {
+					return nil, fmt.Errorf("expected encoder name under index %d to be a String", idx)
+				}
+				encoderName = name.Underlying
+			} else if idx == a.config.EncoderConfigObservationIndex {
+				// extract encoder config from observation
+				conf, ok := highestObservation.(*values.Map)
+				if !ok {
+					return nil, fmt.Errorf("expected encoder config under index %d to be a Map", idx)
+				}
+				encoderConfig = values.ProtoMap(conf)
+			}
+		}
+	}
+	valMap, err := values.NewMap(outcome)
+	if err != nil {
+		return nil, err
+	}
+	return &types.AggregationOutcome{
+		EncodableOutcome: values.ProtoMap(valMap),
```

Comment:
Alternatively, you can make this a value.Value.  There's no reason someone can't encode a number, a list of numbers etc.

Reply:
We actually assume that it needs to be a maps since we add metadata to it.